test/__tests__/plugin-resolution.js

```diff
@@ -0,0 +1,82 @@
+import { runCli } from "../utils";
+
+describe("loads --plugin by its relative path", () => {
+  runCli("plugins", [
+    "automatic/file.txt",
+    "--parser=bar",
+    "--plugin=./automatic/node_modules/prettier-plugin-bar/index.js",
+  ]).test({
+    stdout: "content from `prettier-plugin-bar` package + contents",
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+// TODO (43081j): we don't currently support loading relative paths without
+// the leading `./`
+describe.skip("loads --plugin by its relative path without leading ./", () => {
+  runCli("plugins", [
+    "automatic/file.txt",
+    "--parser=bar",
+    "--plugin=automatic/node_modules/prettier-plugin-bar/index.js",
+  ]).test({
+    stdout: "content from `prettier-plugin-bar` package + contents",
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+describe("loads --plugin by package name", () => {
+  runCli("plugins/automatic", [
+    "file.txt",
+    "--parser=foobar",
+    "--plugin=@user/prettier-plugin-foobar",
+  ]).test({
+    stdout: "foobar+contents",
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+// TODO (43081j): we don't currently support loading relative paths without
+// the leading `./`
+describe.skip("loads --plugin by filename without leading ./, should resolve to file, not package", () => {
```

Comment:
note these skipped tests too

prettier somehow allows `foo/bar/baz.js` to behave as if it was `./foo/bar/baz.js`

no clue how it does this without just trying twice (once for a relative path, once for a node path, because it could also be from the package `foo`)

Reply:
https://github.com/prettier/prettier/blob/af6e7215ce0e0d243cb34a85174af65ab4177f47/src/main/plugins/load-plugin.js#L20-L26

Reply:
aha thank you. so it is what i expected

the question is, do we want this in the new CLI? its easy to add but maybe it'd be a good time to get rid of these relative-but-not-really-relative paths? and just tell people to use actual paths, like `./foo`

Reply:
Actually, I like how it works. Imagine if we support ignore files in config.

```diff
export default {
  ignore: [
-    "foo.js",
+    "./foo.js"
  ]
}
```

No body want this.


Same here

```diff
export default {
  ignore: [
-    "node_modules/prettier-plugin-foo/index.js",
+    "./node_modules/prettier-plugin-foo/index.js"
  ]
}
```

much easier to use.

Mayb someday, we should ask user to import plugin like ESLint did.

Reply:
Add the plugin path also not related to config file, but related to CWD, makes no sense to put `./` there.

Reply:
Forgot details, but tricky part is here https://github.com/prettier/prettier/blob/af6e7215ce0e0d243cb34a85174af65ab4177f47/src/config/resolve-config.js#L67